flow/src/main/scala/org/alephium/flow/core/FlowUtils.scala

```diff
@@ -208,10 +209,20 @@ trait FlowUtils
         templateTs,
         miner
       )
-    } yield template
+      validated <- validateTemplate(chainIndex, template)
+    } yield {
+      if (validated) {
+        template
+      } else {
+        logger.warn("Assemble empty block due to invalid txs")
+        val coinbaseTx =
+          Transaction.coinbase(chainIndex, AVector.empty[Transaction], miner, target, templateTs)
+        template.copy(transactions = AVector(coinbaseTx)) // fall back to empty block
```

Comment:
Better to mine empty block and get the block rewards in my opinion. The extra check is just in case there is a bug in the system. Ideally, we don't need to check